[libcups] Init integration

fish98 opened the pull request:
The OpenPrinting Team is integrating C-based projects into the OSS-Fuzz Framework. The maintaining fuzz harnesses locate in OpenPrinting project [fuzzing](https://github.com/OpenPrinting/fuzzing). 

This PR contains the initial integration config for OpenPrinting project libcups.

CC: @tillkamppeter @iosifache

iosifache commented:
Hi OSS-Fuzz team,

Please note that this PR is part of OpenPrinting's GSoC initiative for integrating C-based repositories into OSS-Fuzz. You can find more details [here](https://wiki.linuxfoundation.org/gsoc/google-summer-code-2024-openprinting-projects#integrating_c-based_openprinting_projects_in_oss-fuzz_testing).

Many thanks!

fish98 commented:
> Hi. It looks like the build is broken. We don't usually merge new projects until their build passes. Please fix.

Hi, thank for your assistance. I have fixed the building for all sanitizers with libfuzzer. For now I have commented the i386 arch and the other two fuzz engine. We will update the configuration after the building setting for libcups project.

jonathanmetzman commented:
> > Hi. It looks like the build is broken. We don't usually merge new projects until their build passes. Please fix.
> 
> Hi, thank for your assistance. I have fixed the building for all sanitizers with libfuzzer. For now I have commented the i386 arch and the other two fuzz engine. We will update the configuration after the building setting for libcups project.

Definitely don't need i386. It's a bonus platform.

fish98 commented:
> > > Hi. It looks like the build is broken. We don't usually merge new projects until their build passes. Please fix.
> > 
> > 
> > Hi, thank for your assistance. I have fixed the building for all sanitizers with libfuzzer. For now I have commented the i386 arch and the other two fuzz engine. We will update the configuration after the building setting for libcups project.
> 
> Definitely don't need i386. It's a bonus platform.

Thanks for your suggestion. I have revised the build script and squashed the commit history. PTAL

iosifache commented:
Thank you for the assistance, @jonathanmetzman!
